### PR TITLE
Embed a calendar with all the events on the Calendar page

### PR DESCRIPTION
**Pull request by MridulS**
If someone opens up the Calendar page they will see a bunch of download link for .ics file but they wouldn't really know what kind of "commitment" they are looking at. This embeds a calendar on the page using the `fullcalendar` package. Everything is loaded via a CDN so no need to compile anything.

### CODE REVIEW

**Comment by MridulS** [comment]
Quick link to the page: https://deploy-preview-252--scientific-python-org.netlify.app/calendars/

**Review by tupui — COMMENTED** [review]
Good idea 👍 

Just a few remarks:

* Buttons do not render properly on small screens (too big and weird alignment)
* It's still difficult to know what is what. Adding colour or a selector for the calendars would help.

**Review by rossbar — COMMENTED** [review]
Great idea, thanks @MridulS this was really high on my wishlist :)

I'm +1 for this and I don't want to derail the discussion, but I'd even advocate to have the calendar on the main landing page for scientific-python.org! IMO it'd be nice to have a summary of community event info in a much more visible place.

But let's focus on getting it in first!

**Comment by MridulS** [comment]
I have added individual colors to the calendar but they are hard coded right now (I don't like that!). It seems like color isn't in the standard of ics-py so I'm not sure if we should add a color property to the yaml definition of a calendar event itself yet.

> Buttons do not render properly on small screens (too big and weird alignment)

There doesn't seem to be a native way of making fullcalendar responsive, unlike google calendar. I could maybe remove this if someone opens this on a small screen?

File: layouts/calendars/list.html [hunk 1]
```diff
@@ -36,6 +36,46 @@ <h5 class="subtitle is-5 is-muted">{{ .Params.Subtitle }}</h5>
...
+                  right: "dayGridMonth,timeGridWeek,timeGridDay",
+                },
+                eventSources: [
+                {{ range $key, $value := $.Site.Data.calendars }}
+                    {
+                        url: '{{ $key }}.ics',
+                        format: 'ics',
+                        backgroundColor: cal_colors.shift(),
+                    },
+                {{ end }}
```
Comment (tupui) [inline layouts/calendars/list.html:65]: I am not sure about the syntax please double check 😅
```suggestion
                {{ range $key, $value := $.Site.Data.calendars }}
                    let cal_color = cal_colors.shift()
                    {
                        url: '{{ $key }}.ics',
                        format: 'ics',
                        backgroundColor: cal_color,
                        border-color: cal_color;
                    },
                {{ end }}
```

**Review by tupui — APPROVED** [review]
Thanks for the update @MridulS. Approving now to keep the ball rolling and we can iterate later.

**Review by rossbar — COMMENTED** [review]
Looking at the calendar preview, it looks like some of the events are respecting a timezone whereas others are not. For example, it looks like NumPy is getting my timezone correct but NetworkX is not - whether this is a problem with the calendar widget or the yaml specifications themselves I'm not sure...

FWIW the easiest way to see this is to switch to the "Week" view

**Comment by MridulS** [comment]
Made some aesthetic changes to the Calendar UI.
@tupui Border color matches the background color.

@rossbar The timezone issue does seem to be a thing with fullcalendar (i think?), maybe it's not picking up the timezone correctly from the ics feed of networkx. [networkx.yaml](https://github.com/scientific-python/scientific-python.org/blob/main/calendars/networkx.yaml) uses `timezone` to declare the timezone and [numpy.yaml](https://github.com/scientific-python/scientific-python.org/blob/main/calendars/numpy.yaml) uses explicit UTC offsets in the begin/end date of the event.

**Comment by rossbar** [comment]
> The timezone issue does seem to be a thing with fullcalendar (i think?), maybe it's not picking up the timezone correctly from the ics feed of networkx.

Hmm... google calendar seems to get it right as I'm subscribed via URL to both numpy and networkx and both show up at the appropriate times in google calendar. Maybe that has something to do with my calendar client settings though :shrug:

**Review by tupui — COMMENTED** [review]
Thanks for the update @MridulS. I would leave the timezone issue as a follow up. But the easy fix is then to remove the timezone indication and convert to UTC.

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**